src/components/Editor.jsx

```diff
@@ -136,12 +136,60 @@ class Editor extends React.Component {
     session.on('change', () => {
       this.props.onInput(this._editor.getValue());
     });
+    session.selection.on('changeCursor', () => {
+      let selector = '';
+      if (this.props.language === 'css') {
+        selector = this._locateSelectorCSS(session);
+      }
+      this._sendSelectorToHighlighter(selector);
+    });
     session.setAnnotations(this.props.errors);
     this._editor.setSession(session);
     this._editor.moveCursorTo(0, 0);
     this._resizeEditor();
   }
 
+  _locateSelectorCSS(session) {
+    const reCssQuery = /(^|.*\})(.*)\{|\}/;
+    let query = false;
+    const lines = session.doc.$lines;
+    const cursor = session.selection.lead;
+    if (!lines[cursor.row]) {
+      return query;
+    }
+    const line = lines[cursor.row].substr(0, cursor.column);
+    let started = false;
+    if (line.match(reCssQuery)) {
+      query = RegExp.$2;
+      started = true;
+    }
+    if (!started || query) {
+      for (let i = cursor.row - 1; i >= 0; i--) {
+        if (started) {
+          if (lines[i].match(/[};]/)) {
+            break;
+          } else {
+            query = `${lines[i]} ${query}`;
+          }
+        } else if (lines[i].match(reCssQuery)) {
+          query = RegExp.$2;
+          if (!query) {
+            break;
+          }
+          started = true;
+        }
+      }
+    }
+    return query;
+  }
+
+  _sendSelectorToHighlighter(selector) {
+    window.frames[0].postMessage(JSON.stringify({
+      type: 'highlightElement',
+      selector: {selector},
+    }), '*');
```

Comment:
So, here we’re bypassing Redux and passing state directly from a React component through the DOM. I am a stickler for materializing all state in the Redux store.

Ergo: I think we need a property somewhere in the Redux store (I don’t have an amazing proposal for where) that holds the current selector; then the preview component should consume that property, watch it for changes, and use `postMessage` to notify the preview frame via `postMessage`.

Reply:
Yes this makes sense!